### PR TITLE
experimental: post results to google sheets

### PR DESCRIPTION
**Pull request by paulirish**
#### Results:
image
image
Live link: https://docs.google.com/spreadsheets/d/1-84NpWXXBNmwkmx4tmWNzR7xnVYRhIBgVxGePA4jsZk/edit?usp=sharing

#### Instructions: 

(There's some overly manual stuff right now...)

1. setup google developer project and get credentials. ([everything in step 1 here](https://developers.google.com/sheets/api/quickstart/nodejs#step_1_turn_on_the_api_name))
2. take the client_secret and place in `lib/gsheets/`
3. run `node lib/gsheets/gsheets.js` and follow instructions for authenicating
4. make a copy of [this spreadsheet](https://docs.google.com/spreadsheets/d/1-84NpWXXBNmwkmx4tmWNzR7xnVYRhIBgVxGePA4jsZk/edit?usp=sharing), remove rows 2-N.
5. copy the ID of the spreadsheet into the `spreadsheetId` in `gsheets.js`

then run `pwmetrics http://paulirish.com --submit`

with luck, you'll be in business.
What's really cool is seeing the chart update the second that new results are posted. :D

### CODE REVIEW

**Comment by denar90** [comment]
Cool feature! 📈 
I have to dig in sheets api to reduce manual stuff. What do you think if we keep it in `sheets` branch for a while? Then can improve this stuff and then merge/release it.

**Comment by paulirish** [comment]
> I have to dig in sheets api to reduce manual stuff. What do you think if we keep it in sheets branch for a while? Then can improve this stuff and then merge/release it.

yup sgtm! im good with it chillin in a branch for a bit. :)

File: readme.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ $ npm install --save pwmetrics
 #              Median run selected by run with the median TTI.
 pwmetrics http://example.com/ --runs=3
 
+# --submit     Submits results to a google spreadsheet (requires some setup)
+pwmetrics http://goat.com --submit
```
Comment (staabm) [inline readme.md]: Do you expect to support other apis besides googlespreadsheet? If so, this option should get a more specific name

Comment (denar90) [inline readme.md]: If so, it will get some value/keyName, I think...

Comment (paulirish) [inline readme.md]: I imagine we'll have a config object somewhere that can have clientsecret/api-token as well as indicate the endpoint.

File: lib/index.js [hunk 2]
```diff
@@ -20,7 +22,6 @@ class PWMetrics {
   constructor(url, opts) {
     this.url = url;
     this.opts = opts;
-    this.metrics = opts.metrics;
```
Comment (denar90) [inline lib/index.js]: Just realized that we should not remove this problem, should we?

**Comment by paulirish** [comment]
btw I merged in master and resolved all the conflicts.

**Comment by paulirish** [comment]
>> @@ -20,7 +22,6 @@ class PWMetrics {
   constructor(url, opts) {
     this.url = url;
     this.opts = opts;
>> -    this.metrics = opts.metrics;

> Just realized that we should not remove this problem, should we?

i don't actually see where `this.metrics` gets used, so.... yeahh... :)

**Comment by denar90** [comment]
Sorry, never mind. Dumb comment, I messed up with merges/branches.

**Comment by MickeyKay** [comment]
Oh man, this is brilliant. Any status update on this feature and when you expect it'll be 1. usable in beta form or 2. released?

**Comment by paulirish** [comment]
Usable in beta form now. :) Just check out this branch and give it a whirl.
If you say it's good enough we can ship it.

**Comment by denar90** [comment]
@MickeyKay if you wanna try it, use `sheets-finalizations` branch :) I think there are latest stuff.

**Pull request merged**